dotCMS/src/main/java/com/dotcms/rest/api/v1/relationships/RelationshipsResource.java

```diff
@@ -0,0 +1,106 @@
+import com.dotcms.util.PaginationUtil;
+import com.dotcms.util.pagination.PaginationException;
+import com.dotcms.util.pagination.RelationshipPaginator;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * This resource provides all the different end-points associated to information and actions that
+ * the front-end can perform on relationships.
+ *
+ * @author nollymar
+ */
+@Path("/v1/relationships")
+public class RelationshipsResource {
+
+    private final WebResource webResource;
+
+    public RelationshipsResource() {
+        this(new WebResource());
+    }
+
+
+    RelationshipsResource(final WebResource webResource) {
+
+        this.webResource = webResource;
+
+    }
+
+    @GET
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response getOneSidedRelationships(
+            @QueryParam("contentTypeId") final String contentTypeId,
+            @QueryParam(PaginationUtil.PAGE) final int page,
+            @QueryParam(PaginationUtil.PER_PAGE) @DefaultValue("-1") final int perPage,
+            @Context final HttpServletRequest request) throws Throwable {
+        Logger.debug(this,
+                "Getting the possible relationships and content types");
+
+        final InitDataObject initData = this.webResource.init(null, true, request, true, null);
+        final User user = initData.getUser();
+        final ContentTypeAPI contentTypeAPI = APILocator.getContentTypeAPI(user);
+        ContentType contentType;
+
+        if (UtilMethods.isSet(contentTypeId)) {
+            //Validate if contentTypeId is valid
+            try {
+                contentType = contentTypeAPI.find(contentTypeId);
+
+            } catch (DotSecurityException | DotDataException e) {
+                return ExceptionMapperUtil
+                        .createResponse(map("message", "Invalid Content Type"),
+                                "Invalid Content Type",
+                                Status.BAD_REQUEST);
+            }
+        } else {
+            return ExceptionMapperUtil
+                    .createResponse(map("message", "Content Type ID is required"),
+                            "Content Type ID is required",
+                            Status.BAD_REQUEST);
+        }
+
+        final PaginationUtil paginationUtil = new PaginationUtil(new RelationshipPaginator(user));
+
+        try {
+
+            final Map<String, Object> params = map(RelationshipPaginator.CONTENT_TYPE_PARAM,
+                    contentType);
+            return paginationUtil.getPage(request, user, null, page, perPage, params);
+        } catch (PaginationException e) {
```

Comment:
there is already something that handle most of the cases for exceptions
return ResponseUtil.mapExceptionResponse(e);